src/services/expenses.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+let expenses = [];
+
+const getAll = (from, to, category, userId) => {
+  if (userId) {
+    let filterExpenses = expenses
+      .filter(expense => expense.userId === +userId);
+
+    if (category) {
+      filterExpenses = filterExpenses
+        .filter(expense => expense.category === category);
+    }
+
+    expenses = filterExpenses;
+  }
+
+  if (from && to) {
+    const filterExpenses = expenses
+      .filter(expense => expense.spentAt > from && expense.spentAt < to);
+
+    expenses = filterExpenses;
+  }
+
+  return expenses;
+};
```

Comment:
Maybe code would be more elegant in you would split this logic into 3 different methods. What do you think?